app/components/MintBTC.tsx

```diff
@@ -4,13 +4,68 @@ import { Link } from "react-router";
 import { Button } from "./ui/button";
 import { FormProvider, useForm } from "react-hook-form";
 import { FormInput } from "./form/FormInput";
-import { useXverseWallet } from "./Wallet/XverseWallet/useWallet";
-import { useContext } from "react";
+import { useXverseConnect, useXverseWallet } from "./Wallet/XverseWallet/useWallet";
+import { useContext, useState } from "react";
 import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { Wallets } from "~/components/Wallet";
 import { FormNumericInput } from "./form/FormNumericInput";
 import { NumericFormat } from "react-number-format";
-import { formatBTC, parseBTC } from "~/lib/denoms";
+import { BTC, formatBTC, parseBTC } from "~/lib/denoms";
+import { nBTCMintTxn } from "~/lib/nbtc";
+import { networks } from "bitcoinjs-lib";
+import { useToast } from "~/hooks/use-toast";
+import { Modal } from "./ui/dialog";
+import { Check } from "lucide-react";
+import { classNames } from "~/util/tailwind";
+import { isValidSuiAddress } from "@mysten/sui/utils";
+import { isMainNetNetwork } from "~/lib/appenv";
+
+interface TransactionStatusProps {
+	SUIAddress: string;
+	txnId: string;
+	handleRetry: () => void;
+}
+
+function TransactionStatus({ SUIAddress, txnId, handleRetry }: TransactionStatusProps) {
+	return (
+		<div className="p-4 rounded-lg text-white flex flex-col gap-4">
+			<div className="flex flex-col items-center gap-2">
+				<Check
+					className={classNames({
+						"text-green-500": true,
+					})}
+					size={30}
+				/>{" "}
+				Success
+				<Link
+					target="_blank"
+					to={`https://mempool.space/testnet4/tx/${txnId}`}
```

Comment:
hardcoded links, what if we want to run it on mainnet, or testnet3

Reply:
Done